packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -570,9 +552,7 @@ export function makeVatKeeper(
     if (snapStore) {
       snapStore.deleteVatSnapshots(vatID);
     }
-
-    const endPos = getRequired(`${vatID}.t.endPosition`);
-    kvStore.set(`${vatID}.t.startPosition`, endPos);
+    transcriptStore.rolloverSpan(vatID);
```

Comment:
Vat termination should be allowed to fully delete the vat data: all transcripts, all heap snapshots, plus of course all the current-moment things like c-lists. Their vatIDs should *also* be removed from the `dynamicIDs` list.

I think we're not fully clear on retaining/not-retaining historical spans for terminated vats (and we haven't yet added an `openSwingStore()` option to retain/not-retain old data). One coherent stance would be for a "vat is terminated" `transcriptStore` API to close off the current span, mark it as not-current in the same way that this `rolloverSpan` API would have done (thus leaving *all* spans marked as not-current), and make sure that whatever DB record is used to decide that certain spans are required artifacts knows to not require one for the terminated vat.

Another coherent stance would be for this "vat is terminated" API to just delete everything immediately.

Reply:
I'm confused by @mhofman's comment.  `rolloverSpan` is used every time there's a snapshot.

Reply:
There is no snapshot when terminating a vat. The last span should be marked inactive on vat termination.

Reply:
That makes sense.  I'm a little nervous about deleting everything on vat termination because it seems like the transcript leading up to a failure would be critical to diagnosing the problem that caused the failure (or is that already adequately captured by the slog?).  So I guess the question is: mark the span as inactive or remove it?

Reply:
I'd say mark the span as inactive, which makes it eligible for pruning based on (future) endpoint configuration. An archive node would keep the spans and could be used for diagnosis.

Reply:
Re-opening this conversation as I don't believe my blocking concern has been addressed: on vat termination, aka when `cleanupAfterTerminatedVat` is executed, we do not seem to close the last transcript span.

Reply:
Since the transcript is now removed, the original issue can be considered addressed.